zarr/storage.py

```diff
@@ -1779,7 +1779,13 @@ def __setitem__(self, key, value):
             self.zf.writestr(keyinfo, value)
 
     def __delitem__(self, key):
-        raise NotImplementedError
+        # raise NotImplementedError
+        try:
+            value = self[key]
+        except KeyError:
+            raise KeyError("Can not delete a nonexistent key")
```

Comment:
Would suggest just checking if we have `key` (instead of reading it in). That way if it is a large value, we don't pay the cost of reading. `__contains__` has [some useful code]( https://github.com/zarr-developers/zarr-python/blob/2cfee9c0f0b2e0782f942ff0dd8681ecfb756ca0/zarr/storage.py#L1808-L1809 ) that we may borrow here.

Reply:
Also done